pkg/sharings/send_mails.go

```diff
@@ -154,20 +154,42 @@ func generateOAuthQueryString(s *Sharing, r *Recipient) (string, error) {
 		return "", err
 	}
 
-	// We use url.encode to safely escape the query string.
-	mapParamQueryString := url.Values{
+	baseURL := r.URL + "/sharings/request"
+	// The link/button we put in the email has to have an http:// or https://
+	// prefix, otherwise it cannot be open in the browser.
+	baseURL = setHTTPPrefixToURL(baseURL, scheme)
+
+	oAuthQuery, err := url.Parse(baseURL)
+	if err != nil {
+		return "", err
+	}
+	// We use url.encode to safely escape the query parameters.
```

Comment:
If you wait until after url.Parse, you dont need to manipulate the url string as much.

if(oAuthQUery.Scheme != "http" && oauthQuery.Scheme != "https") { oauthQuery.Scheme = scheme }